server/protocols/webadmin/webadmin-data/src/main/java/org/apache/james/webadmin/routes/DLPConfigurationRoutes.java

```diff
@@ -0,0 +1,226 @@
+import static org.apache.james.webadmin.Constants.EMPTY_BODY;
+import static org.apache.james.webadmin.Constants.JSON_CONTENT_TYPE;
+import static org.apache.james.webadmin.Constants.SEPARATOR;
+import static org.apache.james.webadmin.dto.DLPConfigurationItemDTO.toDLPConfigurations;
+import static org.apache.james.webadmin.dto.DLPConfigurationItemDTO.toDTOs;
+
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+
+import org.apache.james.core.Domain;
+import org.apache.james.dlp.api.DLPConfigurationItem;
+import org.apache.james.dlp.api.DLPConfigurationStore;
+import org.apache.james.domainlist.api.DomainList;
+import org.apache.james.domainlist.api.DomainListException;
+import org.apache.james.webadmin.Routes;
+import org.apache.james.webadmin.dto.DLPConfigurationItemDTO;
+import org.apache.james.webadmin.utils.ErrorResponder;
+import org.apache.james.webadmin.utils.ErrorResponder.ErrorType;
+import org.apache.james.webadmin.utils.JsonExtractException;
+import org.apache.james.webadmin.utils.JsonExtractor;
+import org.apache.james.webadmin.utils.JsonTransformer;
+import org.eclipse.jetty.http.HttpStatus;
+
+import com.github.steveash.guavate.Guavate;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiImplicitParam;
+import io.swagger.annotations.ApiImplicitParams;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import spark.HaltException;
+import spark.Request;
+import spark.Service;
+
+@Api(tags = "DLPRules")
+@Path(DLPConfigurationRoutes.BASE_PATH)
+@Produces(JSON_CONTENT_TYPE)
+public class DLPConfigurationRoutes implements Routes {
+
+    static final String BASE_PATH = "/dlp/rules";
+
+    private static final String DOMAIN_NAME = ":senderDomain";
+    private static final String SPECIFIC_DLP_RULE_DOMAIN = BASE_PATH + SEPARATOR + DOMAIN_NAME;
+
+    private final JsonTransformer jsonTransformer;
+    private final DLPConfigurationStore dlpConfigurationStore;
+    private final JsonExtractor<DLPConfigurationItemDTO> jsonExtractor;
+    private final DomainList domainList;
+
+    private Service service;
+
+    @Inject
+    public DLPConfigurationRoutes(DLPConfigurationStore dlpConfigurationStore, DomainList domainList, JsonTransformer jsonTransformer) {
+        this.dlpConfigurationStore = dlpConfigurationStore;
+        this.domainList = domainList;
+        this.jsonTransformer = jsonTransformer;
+        this.jsonExtractor = new JsonExtractor<>(DLPConfigurationItemDTO.class);
+    }
+
+    @Override
+    public void define(Service service) {
+        this.service = service;
+
+        defineStore();
+
+        defineList();
+
+        defineClear();
+    }
+
+    @PUT
+    @Path("/{senderDomain}")
+    @ApiOperation(value = "Store a list of dlp configs for given senderDomain")
```

Comment:
Maybe people disagree with me, but I think that we put a config, not a list of configs

Reply:
no I agree, it's a config (for now at list)

Reply:
okay, i will change into singular